Switch project form from tabs to collapsible panels; fixes #630

jdossett opened the pull request:
Lets get rid of tabs and go to collapsible panes. We also now show all fields when javascript is disabled.

This additionally closes #595, #475, and #218. 

Signed-off-by: Jason Dossett <jdossett@utdallas.edu>

david-a-wheeler commented:
On initial look this looks pretty good. However, clearly the panel to open will differ depending on what information has already been provided. This patch always closes the later panels. How about doing a quick check on the 1st question in each panel, and close all the panels by default except the first panel where the first question the is unknown? That way, when people save comma they will by default see the panel that is most likely to be useful.

jdossett commented:
So I just pushed up a number of changes.

1. The url is now updated whenever you open a panel.
2. If you give a panel id in the url (an anchor) the correct panel will open when loading the page.
3. Submit buttons are now at the bottom of each panel.  They also submit the anchor for the section in which they reside. When you save and continue, The panel where you saved will be opened on reload. 
4. Added an expand/collapse all panels button.

Let me know what you think.

david-a-wheeler commented:
Awesome. Lots of pieces here. Let us NOT squash the merge. Then it'll be easier to revert pieces instead of only the whole thing, if we need to do that

david-a-wheeler commented:
A few comments:

* In the panel headings the caret is too close to the heading text.  The CSS should probably be tweaked to add a little margin.
* The buttons and headings are easily confused.  The "Submit" and "Save" buttons currently have the same color as the headings - they should be different.  That should be a quick CSS tweak.  While we're at it, the buttons at the top should also have colors, to make them more obvious (maybe the as save & submit, since they're also buttons?)

david-a-wheeler commented:
Here are some specifics.  Perhaps the buttons at top, like "Expand all panels" and "Show detailed text", should be Bootstrap class "btn-info" instead of "btn-default" (or change the color for btn-default).  That would make it more obvious that they exist.

The CSS classes .btn-primary and .panel-title seem have the same colors, which leads to visual confusion.  We should change the CSS colors, or change a class of one of them.

david-a-wheeler commented:
I worry that because "save and continue" doesn't open the *next* panel, people may not progress.  That's especially true because "save and continue" jumps them up to the top of the panel.  I suggest that "save and continue" open the *next* panel (with wrap around), so that we keep people moving.  They can always uncollapse the previous panel, if they want to.

jdossett commented:
My latest change addresses the button styles.  I think we should discuss more whether save and continue jumps to the next pane.  I can see a scenario where people will save when still working on that pane, but indeed it may be better to go to the next pane.

david-a-wheeler commented:
This doesn't *quite* scroll to the right places in some cases, but we didn't even *have* that functionality earlier.

There are some minor test failures and formatting issues, which I presume you'll fix soon.  Once you fix them, I plan to merge this (not as a squash merge, but as individual commits so we can cherry-pick rollbacks if needed).  That way, the results will be easily visible on the master tier.  We can then fix from there (I think it'd be better to make that a new branch).

Thanks!

jdossett commented:
Yes, I am working on the test failures now.  It is a bit more complicated than I though, but I should have it done by tonight.

jdossett commented:
I reverted the last commit for now.  I am working on an alternative, cleaner approach.

david-a-wheeler commented:
Okay.  I won't merge until you say you're ready.

jdossett commented:
Finally fixed all the little quirks:

Last push does the following:

 * Panel's anchor tag opens that panel on load
 * Child element's anchor tag opens parent panel on load
   and scrolls to childs anchor
 * Save (and continue) opens next panel on reload.
 * css to resolve anchor issues with fixed header

jdossett commented:
There is a problem after the merge from master with the new css.  Somehow clickable areas of the panel headings have resized strangely.

jdossett commented:
Issues fixed, not perfect without js (anchors are slightly off on page load) but with javascript it works perfect.

Ready to pull.

david-a-wheeler commented:
That's fine. Thanks for working this out!
